### PR TITLE
MVC functional test cleanup (part 1)

### PR DESCRIPTION
**Pull request by dougbu**
- #3612 subpart 2 of 2
- merge `ActionConstraintsWebSite`, `AntiforgeryTokenWebSite` and `ContentNegotiationWebSite` into `BasicWebSite`
  - remove tests depending on `ErrorReporterMiddleware` in merged sites; not used in `BasicWebSite`
- delete `CompositeViewEngineWebSite`, `Microsoft.AspNet.Mvc.ViewEngines.CompositeViewEngineTest` covers cases
- delete `CorsMiddlewareWebsite`, soon to be in Cors repo (see aspnet/CORS#57)
- remove `SendFileMiddleware` from FilesWebSite, unused there

### CODE REVIEW

File: test/Microsoft.AspNet.Mvc.FunctionalTests/AntiforgeryTests.cs [hunk 1]
```diff
@@ -23,7 +23,7 @@ public AntiforgeryTests(MvcTestFixture<AntiforgeryTokenWebSite.Startup> fixture)
         public async Task MultipleAFTokensWithinTheSamePage_GeneratesASingleCookieToken()
         {
             // Arrange & Act
-            var response = await Client.GetAsync("http://localhost/Account/Login");
+            var response = await Client.GetAsync("http://localhost/Antiforgery/Login");
```
Comment (dougbu) [inline test/Microsoft.AspNet.Mvc.FunctionalTests/AntiforgeryTests.cs]: Renamed this controller as part of the move.

File: test/Microsoft.AspNet.Mvc.FunctionalTests/ContentNegotiationTest.cs [hunk 2]
```diff
@@ -112,11 +114,11 @@ public ContentNegotiationTest(MvcTestFixture<ContentNegotiationWebSite.Startup>
...
             var expectedContentType = MediaTypeHeaderValue.Parse("application/xml;charset=utf-8");
             var expectedOutput = "<User xmlns:i=\"http://www.w3.org/2001/XMLSchema-instance\" " +
-                                "xmlns=\"http://schemas.datacontract.org/2004/07/ContentNegotiationWebSite\">" +
-                                "<Address>One Microsoft Way</Address><Name>John</Name></User>";
+                "xmlns=\"http://schemas.datacontract.org/2004/07/BasicWebSite.Models\">" +
+                "<Address>One Microsoft Way</Address><Name>John</Name></User>";
             var request = new HttpRequestMessage(
                 HttpMethod.Get,
-                "http://localhost/Home/UserInfo_ProducesWithTypeAndContentType");
+                "http://localhost/ContentNegotiation/UserInfo_ProducesWithTypeAndContentType");
```
Comment (dougbu) [inline test/Microsoft.AspNet.Mvc.FunctionalTests/ContentNegotiationTest.cs]: Another controller rename. `BasicWebSite` already contains a `HomeController`.

File: test/Microsoft.AspNet.Mvc.FunctionalTests/ContentNegotiationTest.cs [hunk 3]
```diff
@@ -287,7 +290,7 @@ public ContentNegotiationTest(MvcTestFixture<ContentNegotiationWebSite.Startup>
             var expectedBody = "{\"MethodName\":\"Produces_WithNonObjectResult\"}";
 
             // Act
-            var response = await Client.GetAsync("http://localhost/JsonResult/Produces_WithNonObjectResult");
+            var response = await Client.GetAsync("http://localhost/ProducesJson/Produces_WithNonObjectResult");
```
Comment (dougbu) [inline test/Microsoft.AspNet.Mvc.FunctionalTests/ContentNegotiationTest.cs]: Another controller rename. `BasicWebSite` already contains a `JsonResultController`.

**Comment by rynowak** [comment]
:shipit:

**Comment by dougbu** [comment]
59824dc w/ minor touch-up in 900663b

**Pull request closed without merging**